
<html>
  <head>
    <title>Lab 05 - Graded Solution</title>
  </head>
  <body>
    <h1>Lab 05 - Graded Solution</h1>
    <p><b>You scored 0% = 0.0 points for this lab.</b></p>
    <ul>
      <li><b>test_subscript_access</b>
      <ul>
      <li>Grade Percentage: 11</li>
      <li>Your solution failed this test case! taking 0.070 sec</Li>
<li><pre>AssertionError: Expected 1 but got None</pre></li>
<li><pre>Traceback (most recent call last):
  File "/Users/lord_pretzel/.pyenv/versions/3.9.0/lib/python3.9/site-packages/grader-0.0.1-py3.9.egg/grader/execution_base.py", line 60, in call_test_function
    test_function(
  File "/Users/lord_pretzel/.pyenv/versions/3.9.0/lib/python3.9/site-packages/grader-0.0.1-py3.9.egg/grader/core.py", line 48, in wrapper
    result = test_function(module, *args, **kwargs)
  File "/var/folders/f7/xnz8v_b13td37s3k4w8t5j2h0000gn/T/tmpsfkbwu66/lab05-test.py", line 17, in test_subscript_access
    assertEquals(lst[i], data[i])
  File "/Users/lord_pretzel/.pyenv/versions/3.9.0/lib/python3.9/site-packages/grader-0.0.1-py3.9.egg/grader/assertions.py", line 58, in assertEquals
    raise AssertionError(message)
AssertionError: Expected 1 but got None
</pre></li>
      </ul>
      </li>
      <li><b>test_custor_based_access</b>
      <ul>
      <li>Grade Percentage: 12</li>
      <li>Your solution failed this test case! taking 0.072 sec</Li>
<li><pre>AssertionError: </pre></li>
<li><pre>Traceback (most recent call last):
  File "/Users/lord_pretzel/.pyenv/versions/3.9.0/lib/python3.9/site-packages/grader-0.0.1-py3.9.egg/grader/execution_base.py", line 60, in call_test_function
    test_function(
  File "/Users/lord_pretzel/.pyenv/versions/3.9.0/lib/python3.9/site-packages/grader-0.0.1-py3.9.egg/grader/core.py", line 48, in wrapper
    result = test_function(module, *args, **kwargs)
  File "/var/folders/f7/xnz8v_b13td37s3k4w8t5j2h0000gn/T/tmp78mv9jwi/lab05-test.py", line 77, in test_custor_based_access
    assert len(lst1) == len(lst2)
AssertionError
</pre></li>
      </ul>
      </li>
      <li><b>test_stringification</b>
      <ul>
      <li>Grade Percentage: 11</li>
      <li>Your solution failed this test case! taking 0.092 sec</Li>
<li><pre>TypeError: __str__ returned non-string (type NoneType)</pre></li>
<li><pre>Traceback (most recent call last):
  File "/Users/lord_pretzel/.pyenv/versions/3.9.0/lib/python3.9/site-packages/grader-0.0.1-py3.9.egg/grader/execution_base.py", line 60, in call_test_function
    test_function(
  File "/Users/lord_pretzel/.pyenv/versions/3.9.0/lib/python3.9/site-packages/grader-0.0.1-py3.9.egg/grader/core.py", line 48, in wrapper
    result = test_function(module, *args, **kwargs)
  File "/var/folders/f7/xnz8v_b13td37s3k4w8t5j2h0000gn/T/tmpm64wu7fk/lab05-test.py", line 129, in test_stringification
    assertEquals('[]', str(lst))
TypeError: __str__ returned non-string (type NoneType)
</pre></li>
      </ul>
      </li>
      <li><b>test_single_element_manipulation</b>
      <ul>
      <li>Grade Percentage: 11</li>
      <li>Your solution failed this test case! taking 0.071 sec</Li>
<li><pre>AssertionError: Expected None but got 849</pre></li>
<li><pre>Traceback (most recent call last):
  File "/Users/lord_pretzel/.pyenv/versions/3.9.0/lib/python3.9/site-packages/grader-0.0.1-py3.9.egg/grader/execution_base.py", line 60, in call_test_function
    test_function(
  File "/Users/lord_pretzel/.pyenv/versions/3.9.0/lib/python3.9/site-packages/grader-0.0.1-py3.9.egg/grader/core.py", line 48, in wrapper
    result = test_function(module, *args, **kwargs)
  File "/var/folders/f7/xnz8v_b13td37s3k4w8t5j2h0000gn/T/tmpxpuoeuhi/lab05-test.py", line 159, in test_single_element_manipulation
    assertEquals(data[i], lst[i])
  File "/Users/lord_pretzel/.pyenv/versions/3.9.0/lib/python3.9/site-packages/grader-0.0.1-py3.9.egg/grader/assertions.py", line 58, in assertEquals
    raise AssertionError(message)
AssertionError: Expected None but got 849
</pre></li>
      </ul>
      </li>
      <li><b>test_predicates</b>
      <ul>
      <li>Grade Percentage: 11</li>
      <li>Your solution failed this test case! taking 0.089 sec</Li>
<li><pre>TypeError: __str__ returned non-string (type NoneType)</pre></li>
<li><pre>Traceback (most recent call last):
  File "/Users/lord_pretzel/.pyenv/versions/3.9.0/lib/python3.9/site-packages/grader-0.0.1-py3.9.egg/grader/execution_base.py", line 60, in call_test_function
    test_function(
  File "/Users/lord_pretzel/.pyenv/versions/3.9.0/lib/python3.9/site-packages/grader-0.0.1-py3.9.egg/grader/core.py", line 48, in wrapper
    result = test_function(module, *args, **kwargs)
  File "/var/folders/f7/xnz8v_b13td37s3k4w8t5j2h0000gn/T/tmpa36qgphk/lab05-test.py", line 189, in test_predicates
    assertEquals(lst, lst2)
  File "/Users/lord_pretzel/.pyenv/versions/3.9.0/lib/python3.9/site-packages/grader-0.0.1-py3.9.egg/grader/assertions.py", line 57, in assertEquals
    message = template.format(got=got, expected=expected, **kw)
TypeError: __str__ returned non-string (type NoneType)
</pre></li>
      </ul>
      </li>
      <li><b>test_queries</b>
      <ul>
      <li>Grade Percentage: 11</li>
      <li>Your solution failed this test case! taking 0.091 sec</Li>
<li><pre>AssertionError: Expected None but got 0</pre></li>
<li><pre>Traceback (most recent call last):
  File "/Users/lord_pretzel/.pyenv/versions/3.9.0/lib/python3.9/site-packages/grader-0.0.1-py3.9.egg/grader/execution_base.py", line 60, in call_test_function
    test_function(
  File "/Users/lord_pretzel/.pyenv/versions/3.9.0/lib/python3.9/site-packages/grader-0.0.1-py3.9.egg/grader/core.py", line 48, in wrapper
    result = test_function(module, *args, **kwargs)
  File "/var/folders/f7/xnz8v_b13td37s3k4w8t5j2h0000gn/T/tmpmx7lzgqb/lab05-test.py", line 215, in test_queries
    assertEquals(0, lst.count(1))
  File "/Users/lord_pretzel/.pyenv/versions/3.9.0/lib/python3.9/site-packages/grader-0.0.1-py3.9.egg/grader/assertions.py", line 58, in assertEquals
    raise AssertionError(message)
AssertionError: Expected None but got 0
</pre></li>
      </ul>
      </li>
      <li><b>test_bulk_operations</b>
      <ul>
      <li>Grade Percentage: 11</li>
      <li>Your solution failed this test case! taking 0.091 sec</Li>
<li><pre>AssertionError: Expected {expected_type} but got {type(got)}</pre></li>
<li><pre>Traceback (most recent call last):
  File "/Users/lord_pretzel/.pyenv/versions/3.9.0/lib/python3.9/site-packages/grader-0.0.1-py3.9.egg/grader/execution_base.py", line 60, in call_test_function
    test_function(
  File "/Users/lord_pretzel/.pyenv/versions/3.9.0/lib/python3.9/site-packages/grader-0.0.1-py3.9.egg/grader/core.py", line 48, in wrapper
    result = test_function(module, *args, **kwargs)
  File "/var/folders/f7/xnz8v_b13td37s3k4w8t5j2h0000gn/T/tmpuxgpgps6/lab05-test.py", line 256, in test_bulk_operations
    assertIsInstance(lst3, m.module.LinkedList)
  File "/Users/lord_pretzel/.pyenv/versions/3.9.0/lib/python3.9/site-packages/grader-0.0.1-py3.9.egg/grader/assertions.py", line 48, in assertIsInstance
    assertTrue(isinstance(got, expected_type), template)
  File "/Users/lord_pretzel/.pyenv/versions/3.9.0/lib/python3.9/site-packages/grader-0.0.1-py3.9.egg/grader/assertions.py", line 65, in assertTrue
    raise AssertionError(template)
AssertionError: Expected {expected_type} but got {type(got)}
</pre></li>
      </ul>
      </li>
      <li><b>test_iteration</b>
      <ul>
      <li>Grade Percentage: 11</li>
      <li>Your solution failed this test case! taking 0.091 sec</Li>
<li><pre>TypeError: iter() returned non-iterator of type 'NoneType'</pre></li>
<li><pre>Traceback (most recent call last):
  File "/Users/lord_pretzel/.pyenv/versions/3.9.0/lib/python3.9/site-packages/grader-0.0.1-py3.9.egg/grader/execution_base.py", line 60, in call_test_function
    test_function(
  File "/Users/lord_pretzel/.pyenv/versions/3.9.0/lib/python3.9/site-packages/grader-0.0.1-py3.9.egg/grader/core.py", line 48, in wrapper
    result = test_function(module, *args, **kwargs)
  File "/var/folders/f7/xnz8v_b13td37s3k4w8t5j2h0000gn/T/tmpvv7y3iwh/lab05-test.py", line 310, in test_iteration
    assertEquals(data, [x for x in lst])
TypeError: iter() returned non-iterator of type 'NoneType'
</pre></li>
      </ul>
      </li>
      <li><b>test_reverse</b>
      <ul>
      <li>Grade Percentage: 11</li>
      <li>Your solution failed this test case! taking 0.094 sec</Li>
<li><pre>TypeError: 'NoneType' object is not subscriptable</pre></li>
<li><pre>Traceback (most recent call last):
  File "/Users/lord_pretzel/.pyenv/versions/3.9.0/lib/python3.9/site-packages/grader-0.0.1-py3.9.egg/grader/execution_base.py", line 60, in call_test_function
    test_function(
  File "/Users/lord_pretzel/.pyenv/versions/3.9.0/lib/python3.9/site-packages/grader-0.0.1-py3.9.egg/grader/core.py", line 48, in wrapper
    result = test_function(module, *args, **kwargs)
  File "/var/folders/f7/xnz8v_b13td37s3k4w8t5j2h0000gn/T/tmpakj12gd8/lab05-test.py", line 335, in test_reverse
    assertEquals(lst[i], rev[len(data) - i - 1])
TypeError: 'NoneType' object is not subscriptable
</pre></li>
      </ul>
      </li>
    </ul>
  </body>
</html>
